[WIP]Add CI for openEuler 22.03

yankay opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md and developer guide https://git.k8s.io/community/contributors/devel/development.md
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**

> /kind feature

**What this PR does / why we need it**:

Add CI Image for openEuler 22.03

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
Add CI Image for openeuler 22.03
```

yankay commented:
/ok-to-test

yankay commented:
/ok-to-test

yankay commented:
/retest

yankay commented:
HI @vannten @erikjiang @cyclinder

Would you please help to review it

tico88612 commented on test-infra/image-builder/roles/kubevirt-images/defaults/main.yml:
```diff
@@ -124,3 +124,17 @@ images:
     converted: true
     tag: "latest"
+
+  openeuler-2203:
+    filename: openEuler-22.03-LTS-SP4-x86_64.qcow2.xz
+    url: https://mirrors.ocf.berkeley.edu/openeuler/openEuler-22.03-LTS-SP4/virtual_machine_img/x86_64/openEuler-22.03-LTS-SP4-x86_64.qcow2.xz
+    checksum: sha256:2dc9c9d73d172af0443d1a17786ee2e387006b9912fad24b5497ef103ecd7afb
+    converted: true
+    tag: "latest"
```
Do we have a test for OpenEuler-2203?
I don't seem to see anything about OpenEuler-2203 testing?

yankay replied:
Yes, there is currently no CI (may not be needed), but the image has been pushed at: https://quay.io/repository/kubespray/vm-openeuler-2203

VannTen commented:
> "msg": "Failed to create object: b'{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"admission webhook \\\\\"virtualmachine-validator.kubevirt.io\\\\\" denied the request: spec.template.spec.volumes[1].cloudInitNoCloud.userDataBase64.cloudInitNoCloud.userDataBase64 is not a valid base64 value.\",\"reason\":\"Invalid\",\"details\":{\"causes\":[{\"reason\":\"FieldValueInvalid\",\"message\":\"spec.template.spec.volumes[1].cloudInitNoCloud.userDataBase64.cloudInitNoCloud.userDataBase64 is not a valid base64 value.\",\"field\":\"spec.template.spec.volumes[1].cloudInitNoCloud.userDataBase64\"}]},\"code\":422}\\n'", "reason": "Unprocessable Entity

Failure (of the openeuler job) looks relevant
Also, why put it in extended rather than regular runs ? Not stable enough ?

yankay commented:
> > "msg": "Failed to create object: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"admission webhook \\"virtualmachine-validator.kubevirt.io\\" denied the request: spec.template.spec.volumes[1].cloudInitNoCloud.userDataBase64.cloudInitNoCloud.userDataBase64 is not a valid base64 value.","reason":"Invalid","details":{"causes":[{"reason":"FieldValueInvalid","message":"spec.template.spec.volumes[1].cloudInitNoCloud.userDataBase64.cloudInitNoCloud.userDataBase64 is not a valid base64 value.","field":"spec.template.spec.volumes[1].cloudInitNoCloud.userDataBase64"}]},"code":422}\n'", "reason": "Unprocessable Entity
> 
> Failure (of the openeuler job) looks relevant Also, why put it in extended rather than regular runs ? Not stable enough ?

HI @VannTen 

Thank you for your review. I have fixed the bug related to the base64 string in the code. Additionally, I have adjusted the CI configuration back to regular runs; previously, I had set it to extended runs to make the CI process faster.

tico88612 commented:
/kind feature
